WeakAuras/WeakAuras.lua

```diff
@@ -5733,3 +5787,55 @@ function WeakAuras.SafeToNumber(input)
   local nr = tonumber(input)
   return nr and (nr < 2147483648 and nr > -2147483649) and nr or nil
 end
+
+function WeakAuras.SetHistory(uid, data, fromAddon, addon)
+  if uid and data then
+    db.history[uid] = db.history[uid] or {}
+    db.history[uid].data = data
+    db.history[uid].source = fromAddon and "addon" or "import"
+    db.history[uid].addon = fromAddon and (addon or "unknown") or nil
+    db.history[uid].skippedVersions = db.history[uid].skippedVersions or {}
+    db.history[uid].lastUpdate = time()
+  end
+end
+
+function WeakAuras.GetHistory(uid)
+  return uid and db.history[uid]
+end
+
+function WeakAuras.RemoveHistory(uid)
+  if uid then
+    db.history[uid] = nil
+  end
+end
+
+function WeakAuras.SkipVersion(uid, version, skip)
+  local history = WeakAuras.GetHistory(uid)
+  if history then
+    history.skippedVersions[version] = skip
+  end
+end
+
+function WeakAuras.IsVersionSkipped(uid, version)
+  local history = WeakAuras.GetHistory(uid)
+  if history then
+    return history.skippedVersions[version]
+  end
+end
+
+function WeakAuras.RestoreFromHistory(uid)
+  local history = WeakAuras.GetHistory(uid)
+  if history and history.data then
+    WeakAuras.Add(history.data)
+  end
+end
+
+function WeakAuras.ClearOldHistory(daysBack, includeNonDeleted)
```

Comment:
When an aura is deleted, the history is not immediately removed, so that they may undelete if they decide that it shouldn't have been deleted.

A sensible UI to "undelete" auras will need to be developed. I imagine that it will be similar in look and feel to a UI to "import" things which external addons have registered, but which haven't yet been added to the database proper (that feature already exists but it is entirely unused, and is not particularly usable in its current state, since the UI is not very intuitive at all).